### PR TITLE
Added Salsa dual-maintenance document

### PR DESCRIPTION
**Pull request by lvoytek**

### CODE REVIEW

**Comment by utkarsh2102** [comment]
Probably needs a rebase for the last commit or something?

**Comment by cpaelzer** [comment]
There are some generalizations which are not necessarily true. I'll add a few hints inline ...

File: SalsaDualMaintenance.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
+Dual Maintenance with Salsa and git ubuntu
+=======
+
+Some packages for Ubuntu are maintained simultaneously using git ubuntu and [Debian Salsa](https://salsa.debian.org).
```
Comment (cpaelzer) [inline SalsaDualMaintenance.md:4]: Please add something like "this pages outlines the model used in mysql and maybe other packages later on".

File: SalsaDualMaintenance.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+These packages include:
+
+* mysql ([Launchpad](https://launchpad.net/ubuntu/+source/mysql-8.0) | [Salsa](https://salsa.debian.org/mariadb-team/mysql/-/tree/mysql-8.0/ubuntu/devel))
+
```
Comment (cpaelzer) [inline SalsaDualMaintenance.md:9]: Add something like "Other packages with dual maintenance and sharing most of the outlined behavior, but not following it exactly are: qemu , dpdk"

File: SalsaDualMaintenance.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+Overview
+--------
+
+Dual maintenace is done such that git ubuntu acts as the primary source of truth for the Ubuntu package through git ubuntu. Meanwhile, the ubuntu/devel branch on Salsa should be synthesized from git ubuntu, while also having references to the Salsa Debian and Upstream branches.
```
Comment (cpaelzer) [inline SalsaDualMaintenance.md:14]: "This particular style of ..." dual maintenance

File: SalsaDualMaintenance.md (hunk 2 above)
Comment (cpaelzer) [reply SalsaDualMaintenance.md:9]: I'd not yet want to add full descriptions of the different styles, but we could make qemu/dpdk a bullet list and add a one line summary for how they are different, that would be like:

- qemu: Neither git-ubuntu nor salsa are dominant - Long term Ubuntu delta is submitted to Debian and becomes part of build-time procedure which generates `d/control` and changes d/rules behavior. This isn't useful of short-term one off patches, but helps a lot to have less churn on merges.
- dpdk: This is primarily managed in salsa and whenever possible Ubuntu is just a sync. All new changes are proposed and committed to the main branch in salsa and cherry picked from there wherever needed. Later in the life-cycle MRE uploads are distribution specific and happen via git-ubuntu.

**Review by cpaelzer — CHANGES_REQUESTED** [review]
I'd like to see that we make it clear that this model is not "the one and only" it is just one example of many. I hope my suggestions help with that.

File: SalsaDualMaintenance.md (hunk 3 above)
Comment (lvoytek) [reply SalsaDualMaintenance.md]: done, thanks for the feedback

Comment (cpaelzer) [reply SalsaDualMaintenance.md]: Thanks, I'm resolving this conversation - let me know what you think of the other two.

File: SalsaDualMaintenance.md (hunk 2 above)
Comment (lvoytek) [reply SalsaDualMaintenance.md:7]: Added these two in an additional packages area

File: SalsaDualMaintenance.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+Merge Requests
+--------------
+
+### Merge Request in Launchpad
+
+In general, merge requests made internally should be done through Launchpad. Once a Launchpad merge request is approved, the commits can be synthesized for Salsa. The transfer to salsa is only required when updating ubuntu/devel.
+
+### Merge Request in Salsa
+
+When a merge request is made to ubuntu/devel in Salsa the commits should be cherry-picked onto ubuntu/devel in git ubuntu then tested from there. Once the merge request is reviewed in Salsa and confirmed to work in git ubuntu, a new merge request should be created in Launchpad. Once this is approved the commits can then be synthesized back into Salsa. Once the synthesized commits are uploaded the merge request can be manually marked as merged.
```
Comment (basak) [inline SalsaDualMaintenance.md:361]: I'm not sure about this, but let's leave it as-is for now, and deal with it when we get there.

File: SalsaDualMaintenance.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    Neither git-ubuntu nor salsa are dominant for qemu - Long term Ubuntu delta is submitted to Debian and becomes part of build-time procedure which generates d/control and changes d/rules behavior. This isn't useful of short-term one off patches, but helps a lot to have less churn on merges.
+
+* dpdk ([Launchpad](https://launchpad.net/ubuntu/+source/dpdk) | [Salsa](https://salsa.debian.org/debian/dpdk))
+
+    This is primarily managed in salsa and whenever possible Ubuntu is just a sync. All new changes are proposed and committed to the main branch in salsa and cherry picked from there wherever needed. Later in the life-cycle MRE uploads are distribution specific and happen via git-ubuntu.
+
+Overview
+--------
+
+This particular style of dual maintenace is done such that git ubuntu acts as the primary source of truth for the Ubuntu package through git ubuntu. Meanwhile, the ubuntu/devel branch on Salsa should be synthesized from git ubuntu, while also having references to the Salsa Debian and Upstream branches.
```
[the remark this answers is not in the document]
Comment (lvoytek) [reply SalsaDualMaintenance.md]: Yeah the wording was I bit weird, updated

File: SalsaDualMaintenance.md [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+
+The tree then becomes:
+
+    upstream:    ---- 1.4
+                         \
+    debian:      -------- 1.4-1 -- logic1 -- logic2 -- logic3
+                                                         |
+    ubuntu/devel:                                        *
+
+Finally, add a commit to the Debian branch updating the changelog to a new version with entries for each logical change. The branch can then be submitted for review in Salsa.
```
[the remark this answers is not in the document]
Comment (lvoytek) [reply SalsaDualMaintenance.md:360]: Alright, I added some more depth on logical changes going into debian

**Comment by lvoytek** [comment]
> Sorry this has taken me so long to get back to. It's not good for it to have dragged on so long. Some minor comments inline. Feel free to adopt or leave as you wish. No need for a further review after this - just go ahead and merge when you're happy please.

Thanks for the review. The changes should be good to go. I don't have write permissions for the repository though

**Pull request merged**